# HG changeset patch
# User Rik <rik@octave.org>
# Date 1501606199 25200
#      Tue Aug 01 09:49:59 2017 -0700
# Node ID 784347c1b7e72471cbb6099e567de5d3c099b4c3
# Parent  d9ca3f15f739bd2f6c38abec074ce882f950bb24
Avoid unnecessary warning when printing hidden axes (bug #51374).

* gl2ps-print.cc (draw_axes): Check whether axes is visible before continuing
to print it.

diff --git a/libinterp/corefcn/gl2ps-print.cc b/libinterp/corefcn/gl2ps-print.cc
--- a/libinterp/corefcn/gl2ps-print.cc
+++ b/libinterp/corefcn/gl2ps-print.cc
@@ -91,16 +91,19 @@ namespace octave
     Matrix render_text (const std::string& txt,
                         double x, double y, double z,
                         int halign, int valign, double rotation = 0.0);
 
     void set_font (const base_properties& props);
 
     void draw_axes (const axes::properties& props)
     {
+      if (! props.is_visible ())
+        return;
+
       // Initialize a sorting tree (viewport) in gl2ps for each axes
       GLint vp[4];
       glGetIntegerv (GL_VIEWPORT, vp);
       gl2psBeginViewport (vp);
 
       // Draw and finish () or there may primitives missing in the gl2ps output.
       opengl_renderer::draw_axes (props);
       finish ();
